libs/ui/src/lib/assistant/assistant.component.ts

```diff
@@ -473,4 +485,46 @@ export class GfAssistantComponent implements OnChanges, OnDestroy, OnInit {
         takeUntil(this.unsubscribeSubject)
       );
   }
+
+  public holdingComparisonFunction(option, value): boolean {
+    return (
+      option.dataSource === value?.dataSource && option.symbol === value?.symbol
+    );
+  }
+
+  private loadPortfolioHoldings() {
+    this.dataService
+      .fetchPortfolioHoldings({
+        range: 'max'
+      })
+      .pipe(takeUntil(this.unsubscribeSubject))
+      .subscribe(({ holdings }) => {
+        this.holdings = sortBy(holdings, ({ name }) => {
+          return name.toLowerCase();
+        });
+        this.setFormValues();
+      });
+  }
+
+  private setFormValues() {
+    const dataSource = this.user?.settings?.['filters.dataSource'] ?? null;
+    const symbol = this.user?.settings?.['filters.symbol'] ?? null;
+    const selectedHolding = this.holdings.find(
+      (holding) =>
+        (holding.dataSource ?? null) === dataSource &&
```

Comment:
Can you use the explicit form please with `{ return ...; }`

Reply:
also used getAssetProfileIdentifier() here